samcli/lib/logs/formatter.py

```diff
@@ -0,0 +1,220 @@
+        return the formatted string. This can be used to parse and format the events based on context
+        ie. In Lambda Function logs, a formatter may wish to color the "ERROR" keywords red,
+        or highlight a filter keyword separately etc.
+
+        This method takes an iterable as input and returns an iterable. It does not immediately format the event.
+        Instead, it sets up the formatter chain appropriately and returns the iterable. Actual formatting happens
+        only when the iterable is used by the caller.
+
+        Parameters
+        ----------
+        event_iterable : iterable of samcli.lib.logs.event.LogEvent
+            Iterable that returns an object containing information about each log event.
+
+        Returns
+        -------
+        iterable of string
+            Iterable that returns a formatted event as a string.
+        """
+
+        for operation in self.formatter_chain:
+
+            # Make sure the operation has access to certain basic objects like colored
+            partial_op = functools.partial(operation, colored=self.colored)
+            event_iterable = map(partial_op, event_iterable)
+
+        return event_iterable
+
+    @staticmethod
+    def _pretty_print_event(event, colored):
+        """
+        Basic formatter to convert an event object to string
+
+        Parameters
+        ----------
+        event
+        colored
+
+        Returns
+        -------
+
+        """
+        event.timestamp = colored.yellow(event.timestamp)
+        event.log_stream_name = colored.cyan(event.log_stream_name)
+
+        return ' '.join([event.log_stream_name, event.timestamp, event.message])
+
+
+class LambdaLogMsgFormatters(object):
+    """
+    Format logs printed by AWS Lambda functions.
+
+    This class is a collection of static methods that can be used within a formatter chain.
+    """
+
+    @staticmethod
+    def colorize_errors(event, colored):
+        """
+
+        Parameters
+        ----------
+        message
+
+        Returns
+        -------
+
+        """
+
+        nodejs_crash_msg = "Process exited before completing request"
+        timeout_msg = "Task timed out"
+
+        if nodejs_crash_msg in event.message \
+                or timeout_msg in event.message:
+            event.message = colored.red(event.message)
+
+        return event
+
+    @staticmethod
+    def colorize_reports(event, colored):
+        """
+
+        Parameters
+        ----------
+        message
+
+        Returns
+        -------
+
+        """
+
+        if event.message.startswith("START") or event.message.startswith("END") or event.message.startswith("REPORT"):
+            event.message = colored.white(event.message)
+
+        return event
+
+
+class KeywordHighlighter(object):
+
+    def __init__(self, keyword=None):
+        self.keyword = keyword
+
+    def highlight_keywords(self, event, colored):
```

Comment:
Nit: highlight gives the impression that the background of these characters will be colors but instead this is just underlining the characters. Can we update this and the class name to reflect the underlining and not highlighting?

Reply:
that's the intention. `highlight` is generic and doesn't describe specific formatting.